src/Internal/Serialization/ToBytes.purs

```diff
@@ -0,0 +1,93 @@
+module Ctl.Internal.Serialization.ToBytes
+  ( class ToBytes
+  , toBytes
+  , toBytes'
+  ) where
+
+import Ctl.Internal.Serialization.Address (Address)
+import Ctl.Internal.Serialization.Hash (Ed25519KeyHash, ScriptHash)
+import Ctl.Internal.Serialization.Types
+  ( AuxiliaryDataHash
+  , DataHash
+  , GenesisDelegateHash
+  , GenesisHash
+  , NativeScript
+  , PlutusData
+  , PoolMetadataHash
+  , Redeemers
+  , ScriptDataHash
+  , Transaction
+  , TransactionBody
+  , TransactionHash
+  , TransactionOutput
+  , TransactionUnspentOutput
+  , TransactionWitnessSet
+  , Value
+  )
+import Ctl.Internal.Types.CborBytes (CborBytes)
+
+-- NOTE returns cbor encoding for all but hash types, for which it returns raw bytes
+foreign import _toBytes :: forall (a :: Type). a -> CborBytes
+
+class ToBytes a where
+  toBytes' :: a -> CborBytes
```

Comment:
This is not accurate for hash types. https://github.com/Emurgo/cardano-serialization-lib/issues/263

Also, the original issue did not request a class here:

> For toBytes, use Castable from untagged-union to remove the need to call asOneOf

What's the reason for the class? If we're keeping it, we need to make sure it's closed such that no other instances could be made, maybe with an instance chain and an error at the end of it.

Reply:
No need for the class. Castable would work better. Ping me if you find the Castable machinery confusing, I have a grasp of it.